File: camera_block_app/lib/models/show.dart
class Show {
  final String id;
  final String name;
  final String startTime;
  final String endTime;
  final String date;

  Show({
    required this.id,
    required this.name,
    required this.startTime,
    required this.endTime,
    required this.date,
  });

  factory Show.fromJson(Map<String, dynamic> json) {
    return Show(
      id: json['_id'],
      name: json['name'],
      startTime: json['start_time'],
      endTime: json['end_time'],
      date: json['date'],
    );
  }
}

File: camera-blocker-backend/models/Show.js
// models/Show.js
const mongoose = require('mongoose');

const showSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
  },
  start_time: {
    type: String, // 또는 Date 타입도 가능
    required: true,
  },
  end_time: {
    type: String, // 또는 Date 타입도 가능
    required: true,
  },
  date: {
    type: String, // 또는 Date 타입도 가능
    required: true,
  },
}, { timestamps: true });

module.exports = mongoose.model('Show', showSchema);
